ext/ffi/lib.rs

```diff
@@ -260,6 +261,57 @@ enum NativeType {
   Pointer,
   Buffer,
   Function,
+  Struct(Box<[NativeType]>),
+}
+
+impl NativeType {
+  fn get_size(&self) -> Result<usize, AnyError> {
+    Ok(match self {
+      NativeType::Void => 0,
+      NativeType::U8 | NativeType::I8 | NativeType::Bool => 1,
+      NativeType::U16 | NativeType::I16 => 2,
+      NativeType::U32 | NativeType::I32 | NativeType::F32 => 4,
+      NativeType::U64 | NativeType::I64 | NativeType::F64 => 8,
+      NativeType::USize
+      | NativeType::ISize
+      | NativeType::Pointer
+      | NativeType::Function
+      | NativeType::Buffer => std::mem::size_of::<usize>(),
+      NativeType::Struct(_) => self.as_layout()?.size(),
+    })
+  }
+
+  fn as_layout(&self) -> Result<Layout, AnyError> {
+    Ok(match self {
+      NativeType::Void => {
+        return Err(type_error("Void type cannot be used as a struct field"))
+      }
+      NativeType::U8 => Layout::new::<u8>(),
+      NativeType::I8 => Layout::new::<i8>(),
+      NativeType::U16 => Layout::new::<u16>(),
+      NativeType::I16 => Layout::new::<i16>(),
+      NativeType::U32 => Layout::new::<u32>(),
+      NativeType::I32 => Layout::new::<i32>(),
+      NativeType::U64 => Layout::new::<u64>(),
+      NativeType::I64 => Layout::new::<i64>(),
+      NativeType::USize => Layout::new::<usize>(),
+      NativeType::ISize => Layout::new::<isize>(),
+      NativeType::F32 => Layout::new::<f32>(),
+      NativeType::F64 => Layout::new::<f64>(),
+      NativeType::Pointer => Layout::new::<usize>(),
+      NativeType::Buffer => Layout::new::<usize>(),
+      NativeType::Function => Layout::new::<usize>(),
+      NativeType::Struct(fields) => {
+        let mut layout = Layout::from_size_align(0, 1)?;
+        for field in fields.iter() {
+          let (new_layout, _) = layout.extend(field.as_layout()?)?;
+          layout = new_layout;
+        }
+        layout.pad_to_align()
+      }
+      NativeType::Bool => Layout::new::<bool>(),
```

Comment:
I guess bool type could be at the very top, maybe after `Void` since it's semi alphabetical ordered AND byte size ordered.